### DISCUSSION TITLE
Custom help text for FilterSet

### DISCUSSION THREAD

**Discussion by snowman2** [carltongibson/django-filter#1380] [Ideas]
What are your thoughts on adding optional `help_text` to the `FilterSet`. 

Here is an example of how we use it:

```pyhton
class AttributeFilter(FilterSet):
    class Meta(object):
        model = Attribute
        fields = {
            "id": ["exact", "in"],
            "name": ["exact", "iexact", "icontains", "contains"],
        }
        help_text = {
            "id": {
                "exact": "id/primary key for an Attribute entry.",
                "in": "Comma separated list of ids/primary keys for a set of Attribute entries.",
            },
            "name": {
                "exact": "Attribute name field.",
                "iexact": "Case insensitive search for attribute name field.",
                "icontains": "Case insensitive `like` search for Attribute names.",
                "contains": "Case sensitive `like` search for Attribute names.",
            },
        }
```

This is useful when auto-generating API documentation.

cc: @alfredoahds

Note: I should add that we would be willing to submit a PR if you would like to see this added.

**Comment by carltongibson** [comment]
Hi. 

My initial response is against adding extra options to FilterSet here. For me, better to do this with the existing API on the form. Adding that at the higher level just isn't worth the complexity. 

Presumably you customised the metaclass to get this to work?

**Reply by snowman2** [reply]
> For me, better to do this with the existing API on the form.

Mind elaborating a bit on this?

> Presumably you customised the metaclass to get this to work?

Yes, that is correct.

**Reply by snowman2** [reply]
Hmmm, after doing some digging, it seems we are using https://github.com/philipn/django-rest-framework-filters. Maybe it would make more sense to be added there.
